File: news_server/modules/auth/presentation/http/registry.ts
/**
 * Handler 注册表
 * 类似 Sjgz-Backend 的 interfaces/http/registry.go
 */
import { AuthHandler } from './handler/auth'

export class Registry {
  Auth: AuthHandler

  constructor(authHandler: AuthHandler) {
    this.Auth = authHandler
  }
}

File: modules/report/interfaces/http/registry.go
package http

import (
	authconn "nfxnews/connections/auth"
	reportapp "nfxnews/modules/report/application/report"
	"nfxnews/modules/report/interfaces/http/handler"
)

type Registry struct {
	App  *handler.ReportHandler
	I18n *handler.I18nHandler
}

func NewRegistry(svc *reportapp.Service, langs string, identity *authconn.Client) *Registry {
	return &Registry{App: handler.NewReportHandler(svc, identity), I18n: handler.NewI18nHandler(langs)}
}
